strawberryfields/io.py

```diff
@@ -146,6 +176,71 @@ def to_program(bb):
     return prog
 
 
+# pylint:disable=too-many-branches
+def _to_tdm_program(bb):
+    # pylint: disable=import-outside-toplevel
+    from strawberryfields.tdm.tdmprogram import TDMProgram
+
+    prog = TDMProgram(max(bb.modes) + 1, name=bb.name)
+
+    # retrieve all the free parameters in the Blackbird program (e.g. "p0", "p1"
+    # etc.) and add their corresponding values to args
+    args = []
+    for k in bb._var.keys():
+        if k[0] == "p" and k[1:].isdigit():
+            v = bb._var[k].flatten()
+            args.append(v)
+
+    # append the quantum operations
+    with prog.context(*args) as (p, q):
+        for op in bb.operations:
+            # check if operation name is in the list of
+            # defined StrawberryFields operations.
+            # This is used by checking against the ops.py __all__
+            # module attribute, which contains the names
+            # of all defined quantum operations
+            if op["op"] in ops.__all__:
+                # get the quantum operation from the sf.ops module
+                gate = getattr(ops, op["op"])
+            else:
+                raise NameError("Quantum operation {} not defined!".format(op["op"]))
+
+            # create the list of regrefs
+            regrefs = [q[i] for i in op["modes"]]
+
+            if "args" in op:
```

Comment:
Perhaps `"kwargs" in op` would anyways be assumed `if "args" in op`, yet think it's good to keep it explicit (and potentially think if there would be an edge case where it could happen that exactly one is defined).
```suggestion
            if "args" in op and "kwargs" in op:
```